src/lr_autoreduce/reduce_REF_L.py

```
@@ -0,0 +1,213 @@
+    parser.add_argument("template_file", nargs="?", default=None, type=str, help='Path to the template XML file.')
+    parser.add_argument("avg_overlap", nargs="?", default="false")
+    parser.add_argument("const_q", nargs="?", default="false")
+    parser.add_argument("fit_first_peak", nargs="?", default="false")
+    parser.add_argument("theta_offset", nargs="?", default="0")
+    # Optional arguments
+    parser.add_argument('--no_publish', action='store_true', help='Do not upload HTML report to the livedata server.')
+
+    return parser.parse_args()
+
+
+def autoreduce(events_file: str, output_dir: str, template_file: str = None, avg_overlap: bool = False, const_q: bool = False, theta_offset: float = 0.0, publish = True) -> None:
```

Ruff (0.14.8)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)